statsmodels/tsa/holtwinters.py

```diff
@@ -0,0 +1,858 @@
+            parse or a datetime type. However, if the dates index does not
+            have a fixed frequency, end must be an integer index if you
+            want out of sample prediction. Default is the last observation in
+            the sample.        
+
+        Returns
+        -------
+        forecast : array
+            Array of out of sample forecasts.
+        """
+        return self.model.predict(self.params, start, end)
+
+    def forecast(self, steps=1):
+        """
+        Out-of-sample forecasts
+
+        Parameters
+        ----------
+        steps : int
+            The number of out of sample forecasts from the end of the
+            sample.
+
+        Returns
+        -------
+        forecast : array
+            Array of out of sample forecasts
+        """
+        try:
+            start = self.model._index[-1] + 1
+            end = self.model._index[-1] + steps
+            return self.model.predict(self.params, start=start, end=end)
+        except ValueError:
+            # May occur when the index doesn't have a freq
+            return self.model._predict(h=steps, **self.params).fcast
+
+
+class HoltWintersResultsWrapper(ResultsWrapper):
+    _attrs = {'fittedvalues': 'rows',
+              'level': 'rows',
+              'resid': 'rows',
+              'season': 'rows',
+              'slope': 'rows'}
+    _wrap_attrs = union_dicts(ResultsWrapper._wrap_attrs, _attrs)
+    _methods = {'predict': 'dates',
+                'forecast': 'dates'}
+    _wrap_methods = union_dicts(ResultsWrapper._wrap_methods, _methods)
+
+
+populate_wrapper(HoltWintersResultsWrapper, HoltWintersResults)
+
+
+class ExponentialSmoothing(TimeSeriesModel):
+    """
+    Holt Winter's Exponential Smoothing
+
+    Parameters
+    ----------
+    endog : array-like
+        Time series
+    trend : {"add", "mul", "additive", "multiplicative", None}, optional
+        Type of trend component.
+    damped : bool, optional
+        Should the trend component be damped.
+    seasonal : {"add", "mul", "additive", "multiplicative", None}, optional
+        Type of seasonal component.
+    seasonal_periods : int, optional
+        The number of seasons to consider for the holt winters.
+
+    Returns
+    -------
+    results : ExponentialSmoothing class        
+
+    Notes
+    -----
+    This is a full implementation of the holt winters exponential smoothing as
+    per [1]. This includes all the unstable methods as well as the stable methods.
+    The implementation of the library covers the functionality of the R 
+    library as much as possible whilst still being pythonic.
+
+    References
+    ----------
+    [1] Hyndman, Rob J., and George Athanasopoulos. Forecasting: principles and practice. OTexts, 2014.
+    """
+
+    def __init__(self, endog, trend=None, damped=False, seasonal=None,
+                 seasonal_periods=None, dates=None, freq=None, missing='none', **kwargs):
```

Comment:
I think we should remove the `**kwargs` here. AFAICS, they are not used in this case, and just swallow without feedback any incorrect keywords

Reply:
done